src/CachePeers.cc

```diff
@@ -12,13 +12,30 @@
 #include "ConfigOption.h"
 #include "configuration/Smooth.h"
 #include "neighbors.h"
+#include "PeerPoolMgr.h"
 #include "PeerSelectState.h"
 #include "SquidConfig.h"
 
 #include <algorithm>
 
 /* CachePeers */
 
+CachePeers::~CachePeers()
+{
+    auto it = storage.cbegin();
+    while(it != storage.cend()) {
+        it = remove(it);
+    }
```

Comment:
Let's simplify and avoid remove() changes by always removing the first peer. The overhead of find_if() is negligible when finding the first peer... Something along these lines may work:

```suggestion
    while (!storage.empty())
        remove(storage.front());
```

Reply:
Done (c5878a6).